src/run_docker.py

```diff
@@ -157,32 +156,24 @@ async def sleep_until_next_polling(current_polling_duration_s):
         await sleep_until_next_polling(polling_duration)
 
 
-def run_loop_forever():
-    try:
-        loop.run_forever()
-    finally:
-        print("Closing AsyncIO loop...")
-        webserver.close_and_cleanup(loop)
-        loop.close()
-
-
 def main():
+    threading.Thread(target=run_webserver_on_asyncio_loop_forever,
+                     args=(loop,),
+                     name="AioHttpLoopWaiterThread",
+                     daemon=True).start()
+
     print_dynatrace_logo()
 
     logging_context.log("GCP Monitor - Dynatrace integration for Google Cloud Platform monitoring\n")
 
-    webserver.setup_webserver_on_asyncio_loop(loop, HEALTH_CHECK_PORT)
-
-    instance_metadata = loop.run_until_complete(run_instance_metadata_check())
-    loop.run_until_complete(import_self_monitoring_dashboards(instance_metadata))
+    instance_metadata = asyncio.run(run_instance_metadata_check())
```

Comment:
Please make sure that if this check fails then exception is propagated and fails main execution. (assuming this was previous behaviour, at least from my memory)

Reply:
Yes, it maintains that